### PR TITLE
[terra-alert] Update alert notification focus shift for dismiss

### PR DESCRIPTION
**Pull request by trandrew1023**
### Summary

**What was changed:**
Added focus shift for `alert` notifications that contain dismiss buttons.

**Why it was changed:**

Focus should be shifted when the alert contains an interactable element. Previously, the focus shifted when there was an action element only.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

No interactable elements:
image
Only dismiss:
image
Only action:
image
Action and dismiss:
image

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9175 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by mjpalazzo** [comment]
Looks good, Andrew!

File: packages/terra-alert/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Unreleased
 
 * Added
-  * Added focus shift to the rendered notification banners content when notification is an alert and includes an action element
+  * Added focus shift to rendered `alert` notification banners content that include an interactable action element and/or dismiss.
 
 * Changed
   * Locked `uuid` to `8.2.0` to maintain IE compatibility.
```
Comment (sdadn) [inline packages/terra-alert/CHANGELOG.md:9]: This change was already released so can we rebase with main so this PR is up to date?

Comment (trandrew1023) [reply packages/terra-alert/CHANGELOG.md]: Rebased changes with main and updated changelog here https://github.com/cerner/terra-core/pull/3823/files#diff-b62ab7ae03f56c47f8aba033b108058b187c6bb44ff7dbe93a72c454c1393279

File: packages/terra-alert/tests/wdio/alert-spec.js [hunk 2]
```diff
@@ -62,12 +62,25 @@ Terra.describeViewports('Alert', ['tiny', 'large'], () => {
   });
 
   describe('Critical alert notification banner', () => {
-    it('alert content is focused when rendered with an action element', () => {
+    it('alert content is focused when rendered with an action element and dismiss', () => {
       browser.url('/raw/tests/cerner-terra-core-docs/alert/custom-prop-alert');
-
       browser.keys(['Tab', 'Tab', 'Tab', 'Tab', 'Enter']);
 
-      Terra.validates.element('alert focused');
```
Comment (smason0) [inline packages/terra-alert/tests/wdio/alert-spec.js:70]: Can these obsolete reference screenshots be removed?

Comment (trandrew1023) [reply packages/terra-alert/tests/wdio/alert-spec.js:70]: Thanks for catching that, removed here https://github.com/cerner/terra-core/pull/3823/commits/5ff0e554ca3d159a06b6a78b3db69905d321d860

**Review by smason0 — APPROVED** [review]

File: packages/terra-alert/tests/wdio/alert-spec.js [hunk 3]
```diff
@@ -62,12 +62,25 @@ Terra.describeViewports('Alert', ['tiny', 'large'], () => {
...
+      Terra.validates.element('alert focused with action and dismiss');
+    });
+
+    it('alert content is focused when rendered with only action element', () => {
+      browser.url('/raw/tests/cerner-terra-core-docs/alert/custom-prop-alert');
+      browser.keys(['Tab', 'Tab', 'Tab', 'Down arrow', 'Down arrow', 'Down arrow', 'Down arrow', 'Down arrow', 'Down arrow', 'Enter', 'Escape', 'Tab', 'Enter']);
+
+      Terra.validates.element('alert focused with action');
```
Comment (sdadn) [inline packages/terra-alert/tests/wdio/alert-spec.js:76]: Can we explicitly check for focus, e.g:
https://github.com/cerner/terra-framework/blob/18ce46e82c0dc60d1665700a2453bb92ed9df6c7/packages/terra-slide-panel/tests/wdio/slide-panel-spec.js#LL191C14-L191C14

File: packages/terra-alert/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Updated focus shift to rendered `alert` notification banners content that include an interactable action element and/or dismiss.
```
Comment (sycombs) [inline packages/terra-alert/CHANGELOG.md:6]: Can we reword it to something like this?
```suggestion
  * Updated focus shift for `alert` notification banners to include notification banners that have an interactable action element and/or dismiss.
```

File: packages/terra-alert/tests/wdio/alert-spec.js (hunk 3 above)
Comment (trandrew1023) [reply packages/terra-alert/tests/wdio/alert-spec.js:78]: Added focus check here https://github.com/cerner/terra-core/pull/3823/commits/d4700b8f71c7da3a79709a02b3651a0419dcdd71#diff-4310e0b70d7468d01c489f156612886bbb0be04b7579c1bd7bb2583723e68521

File: packages/terra-alert/CHANGELOG.md (hunk 4 above)
Comment (trandrew1023) [reply packages/terra-alert/CHANGELOG.md]: Updated changelog here https://github.com/cerner/terra-core/pull/3823/commits/d4700b8f71c7da3a79709a02b3651a0419dcdd71#diff-b62ab7ae03f56c47f8aba033b108058b187c6bb44ff7dbe93a72c454c1393279

**Review by sdadn — APPROVED** [review]

**Comment by trandrew1023** [comment]
Discussed offline by @chrismichalewicz  @eawww  @brooksienoodlesoup: the user's focus should only be moved when there is an actionable item like an action button. If the only interactable option is to dismiss the alert, unless the alert is covering up/obscuring the user's view of the page content focus does not need to be moved since  the main goal is to not move focus when the only purpose of the alert is to convey information.

Closing the PR since previous functionality will only move focus when there is an action element present and these changes will affect the expected behavior.
